DOCS-1862,DOCS-1857: Add Viam Agent

andf-viam opened the pull request:
Add Viam Agent initial docs
- Add general overview
- Add installation steps
- Add initial provisioning behavior: connect to available configured network, or create own hotspot
- Briefly mentioning pre-install option
- Briefly mentioning subsystem, with `agent-provisioning` as example
- Briefly mentioning using mobile app as way of accessing captive portal, if hotspot

Notes:
- This PR does _not_ capture the Viam network management subsystem, as detailed in DOCS - 1413. That will be a separate PR (although it looks [rolled into](https://github.com/viamrobotics/agent-provisioning/pull/1) the now-is-documented Viam Provisioning subsystem? Not clear at first glance, will investigate for next ticket.)
- Not mentioning per-Linux requirements, since [install script handles this behind the scenes](https://github.com/viamrobotics/agent-provisioning#installation).
- Not sure why `htmltest` failing on GH readme links.

Question:
- Did I capture the sprit of subsystem in the last section on this page? Or is it just how Viam ourselves breaks down functionality, and this isn't for end users to experiment with?

andf-viam commented:
Hi @MirandaKapin & @Otterverse -- here's a quick overview of Viam Agent with some basic usage as modeled on Miranda's  _Device Provisioning and Network Configuration_ doc. Let me know if you'd like to see any changes. Thanks!

ale7714 commented on docs/build/provision/_index.md:
```diff
@@ -0,0 +1,251 @@
+
+Using the Viam Agent, this company could ship their machines directly to customers and have each machine provision `viam-server` as it comes online for each user, eliminating factory setup time and allowing for tailored configurations per customer as needed.
+
+The Viam Agent is [open source](https://github.com/viamrobotics/agent), and is available for Linux machines running the `x86_64` and `arm64` architectures.
```
FYI the Viam Agent is currently private, maybe we can hold on this line until we make it public.

ale7714 commented on docs/build/provision/_index.md:
```diff
@@ -0,0 +1,251 @@
+## Use a subsystem
+
+The Viam Agent support modular plugins known as _subsystems_ to allow you to extend the functionality of the Viam Agent to meet your needs.
+You can use existing subsystems written by Viam, or write your own.
```
We're not quite ready for users to write subsystems yet.

ale7714 reviewed (commented):
It's looking great! Left a couple of comments on lines I think we should hold on mentioning for now.

npentrel commented on docs/build/provision/_index.md:
```diff
@@ -0,0 +1,251 @@
+
+You can also use the [Viam mobile application](/fleet/#the-viam-mobile-app), available for download from the [Apple](https://apps.apple.com/us/app/viam-robotics/id6451424162) and [Google Play](https://play.google.com/store/apps/details?id=com.viam.viammobile&hl=en&gl=US) app stores, to connect to the Viam Agent on deployed machines.
+
+See [Mobile App Provisioning](https://github.com/viamrobotics/agent-provisioning#mobile-app-provisioning) for more information.
```
This is also still private... @nataliajacobowitz can we make this public? Can discuss via slack if needed

nataliajacobowitz replied:
Is it worth just ingesting the info there into here since there isn't much information here and then people don't need to click on a link

nataliajacobowitz replied:
I would just add this instead: 
Within the Viam mobile app once logged in, navigate to an organization, then to a location, then inside any location tap "Add new smart machine", and follow the instructions in app.

npentrel reviewed (approved):
This looks great. I've added some minor feedback but don't think I need to review again. Will leave factual review to SMEs. 

`htmltest` fails because the repos aren't public.

andf-viam commented:
@ale7714 Thanks! Removed links to private repos, and added Natalia's suggestion for one such removal. When repos are public, I'll add "for more info, see" links back in. Let me know if you'd like to see any further changes, else I'll merge this. Thank you!

ale7714 reviewed (approved):
Looks great! 🚀
